client/landing/gutenboarding/hooks/use-on-site-creation.ts

```diff
@@ -54,52 +50,30 @@ export default function useOnSiteCreation(): void {
 			setIsRedirecting( true );
 
 			if ( selectedPlan && ! selectedPlan?.isFree && planProductSource ) {
-				const planProduct: RequestCartProduct = createRequestCartProduct( {
-					product_id: planProductSource.productId,
-					product_slug: planProductSource.storeSlug,
-					extra: {
-						source: 'gutenboarding',
-					},
-				} );
+				const productSlugsToAddToCart: string[] = [ planProductSource.storeSlug ];
 
-				let domainProduct: RequestCartProduct | null = null;
 				if ( domain?.product_id && domain?.product_slug ) {
-					domainProduct = createRequestCartProduct( {
-						meta: domain.domain_name,
-						product_id: domain.product_id,
-						product_slug: domain.product_slug,
-						extra: {
-							privacy: domain.supports_privacy,
```

Comment:
I'm not certain, but I _think_ that this property doesn't have any actual effects, but we should probably double check that.

Reply:
Hm... this property does seem to be set by the domains page also:

https://github.com/Automattic/wp-calypso/blob/39467cea5d88fe61ee55147968e46fea5cfbf6db/client/my-sites/domains/domain-search/index.jsx#L142-L144

But it isn't set if checkout itself adds a domain product. Maybe it _is_ important. I'll take a look to see if I can modify `usePrepareProductsForCart` to add it for domain products.

Reply:
I've added `privacy` in 38f3759 based on `is_privacy_protection_product_purchase_allowed` in the products list.